drivers/filesystems/fastfat/cleanup.c

```diff
@@ -186,11 +187,11 @@ VfatCleanup(
     }
 
     ExAcquireResourceExclusiveLite(&IrpContext->DeviceExt->DirResource, TRUE);
-    Status = VfatCleanupFile(IrpContext);
-    ExReleaseResourceLite(&IrpContext->DeviceExt->DirResource);
+    Deleted = VfatCleanupFile(IrpContext);
+    if (!Deleted) ExReleaseResourceLite(&IrpContext->DeviceExt->DirResource);
 
     IrpContext->Irp->IoStatus.Information = 0;
-    return Status;
+    return STATUS_SUCCESS;
```

Comment:
Question: should we always return STATUS_SUCCESS even if VfatCleanupFile() "failed"?

Reply:
I'm not seeing a code path where the previous VfatCleanupFile could have failed (all two `return` statements returned `STATUS_SUCCESS`), so I guess this is fine.

Reply:
VfatCleanupFile always returned STATUS_SUCCESS previously